scripts/operations/configuration/python_lib/vault.py

```diff
@@ -155,134 +137,153 @@ def get_secret_data_from_response(api_response: requests.models.Response) -> JSO
+        additional_headers = kwargs.pop("headers", None)
+        # Generate request headers, combining additional ones, if any
+        request_headers = self.get_api_request_headers(additional_headers)
+        return api_requests.post_retry_validate(headers=request_headers, **kwargs)
+
+    def delete_secret(self, secret_key: str) -> None:
+        """
+        Deletes the specified secret key from Vault. This is also required in the case where the
+        last field of data in a secret is being removed, because Vault does not allow for empty
+        secrets.
+        Raises an exception if anything goes wrong.
+        """
+        secret_url = self.get_secret_api_url(secret_key)
+        logging.debug(f"{secret_key} secret URL = {secret_url}")
+        logging.debug(f"Deleting {secret_key} secret from Vault")
+        self.api_delete(
+            url=secret_url, expected_status_codes=API_STATUS_OK_EMPTY)
+
+    def get_secret(self, secret_key: str, must_exist: bool = False) -> JSONObject:
+        """
+        Retrieves the Vault secret with the specified key and returns its contents.
+        The must_exist argument governs whether a 404 response status code results
+        in an exception or not. If must_exist is False and the response is 404, then
+        None is returned.
+        An exception is raised if any problems occur
+        """
+        secret_url = self.get_secret_api_url(secret_key)
+        logging.debug(f"{secret_key} secret URL = {secret_url}")
+        if must_exist:
+            expected_status_codes = {API_STATUS_OK_WITH_DATA}
+        else:
+            expected_status_codes = {
+                API_STATUS_OK_WITH_DATA, API_STATUS_NOT_FOUND}
+
+        logging.debug(f"Reading {secret_key} secret from Vault")
+        resp = self.api_get(
+            url=secret_url, expected_status_codes=expected_status_codes)
+        if resp.status_code == API_STATUS_NOT_FOUND:
+            # Since an exception was not raised, this must mean we are in the case where
+            #  this is okay. So return None.
+            return None
+        # Return the secret data from the response
+        return get_secret_data_from_response(resp)
+
+    def write_secret(self, secret_key: str, secret_data: JSONObject) -> None:
+        """
+        Writes the specified secret data to Vault using the specified secret key.
+        Raises an exception if anything goes wrong.
+        """
+        secret_url = self.get_secret_api_url(secret_key)
+        logging.debug(f"{secret_key} secret URL = {secret_url}")
+        logging.debug(f"Writing {secret_key} secret to Vault")
+        self.api_post(
+            url=secret_url, expected_status_codes=API_STATUS_OK_EMPTY, json=secret_data)
+
+    def delete_csm_root_secret(self, **kwargs) -> None:
+        """
+        Wrapper function that supplies the CSM root secret key to delete_secret()
+        """
+        self.delete_secret(secret_key=CSM_ROOT_SECRET_KEY, **kwargs)
```

Comment:
```suggestion
        self.delete_secret(secret_key=CSM_ROOT_SECRET_KEY)
```

Appears to me `delete_secret` won't accept any extra args.